src/test/java/seedu/address/model/internship/InternshipTest.java

```diff
@@ -57,10 +56,6 @@ public void isSameInternship() {
         editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withContactEmail("alicer@exampler.com").build();
         assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
 
-        // different application status, all other attributes same -> returns false
-        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withApplicationStatus("rejected").build();
-        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
-
         // different description, all other attributes same -> returns false
```

Comment:
Please add checks for new identity fields!
// different role, all other attributes same -> returns false
        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withRole("Data Analyst").build();
        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));
        
        // different location, all other attributes same -> returns false
        editedAlice = new InternshipBuilder(ALICE_MICROSOFT).withLocation("overseas").build();
        assertFalse(ALICE_MICROSOFT.isSameInternship(editedAlice));